### PR TITLE
Material datums - Moogle Helped - Check #10169 for more info

### PR DESCRIPTION
**Pull request by JamieD1**
Material datums - Moogle Helped - Check #10169 for more info

Adds TGUI for Mech Fab,
Fixes Stand TGUI,
Etc
#### Changelog

:cl:  
rscadd: Material datums - Moogle Helped - Check #10169 for more info
rscadd: Fixes Stand GUI/Mech Fab TGUI
/:cl:

### CODE REVIEW

**Review by Partheo — APPROVED** [review]

File: code/game/atoms.dm [hunk 1]
```diff
@@ -156,6 +161,15 @@
 	if (canSmoothWith)
 		canSmoothWith = typelist("canSmoothWith", canSmoothWith)
 
+	if(custom_materials && custom_materials.len)
+		var/temp_list = list()
+		for(var/i in custom_materials)
+			var/datum/material/material = getmaterialref(i) || i
+			temp_list[material] = custom_materials[material] //Get the proper instanced version
+
+		custom_materials = null //Null the list to prepare for applying the materials properly
```
Comment (Amelia0010) [inline code/game/atoms.dm:170]: So this sets the list to null before running set_custom_materials()

File: code/game/atoms.dm [hunk 2]
```diff
@@ -1127,3 +1146,18 @@
 		var/atom/atom_orbiter = o
 		output += atom_orbiter.get_all_orbiters(processed, source = FALSE)
 	return output
+
+///Sets the custom materials for an item.
+/atom/proc/set_custom_materials(var/list/materials, multiplier = 1)
+	if(custom_materials) //Only runs if custom materials existed at first. Should usually be the case but check anyways
```
Comment (Amelia0010) [inline code/game/atoms.dm:1152]: And this comment says that the list should usually exist?

File: code/game/machinery/autolathe.dm [hunk 3]
```diff
@@ -333,18 +339,37 @@
...
+		var/list/materials_used = list()
+		var/list/picked_materials
+		var/list/custom_materials = list() //These will apply their material effect, This should usually only be one.
+		for(var/MAT in request.materials)
+			var/datum/material/used_material = MAT
+			var/amount_needed = request.materials[MAT] * coeff * multiplier
+			if(istext(used_material)) //This means its a category
+				var/list/list_to_show = list()
+				for(var/i in SSmaterials.materials_by_category[used_material])
+					if(materials.materials[i] > 0)
```
Comment (Amelia0010) [inline code/game/machinery/autolathe.dm]:
```suggestion
					if(materials.materials[i])
```

File: code/game/machinery/autolathe.dm [hunk 4]
```diff
@@ -367,6 +392,9 @@
 					for(var/mat in materials_used)
 						new_item.materials[mat] = materials_used[mat] / multiplier
 					new_item.autolathe_crafted(src)
+					if(length(picked_materials))
```
Comment (Amelia0010) [inline code/game/machinery/autolathe.dm]:
```suggestion
					if(picked_materials.len)
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
It's probably fine tbh

File: code/game/atoms.dm (hunk 1 above)
Comment (JamieD1) [reply code/game/atoms.dm:170]: Yes it runs it on the temp_list not the custom_materials

File: code/game/atoms.dm (hunk 2 above)
Comment (JamieD1) [reply code/game/atoms.dm:1152]: > And this comment says that the list should usually exist?

It does exist in atoms.dm

**Pull request merged**